physionet-django/project/authorization/access.py

```diff
@@ -0,0 +1,123 @@
+from datetime import datetime
+
+from django.db.models import Q
+
+from events.models import Event, EventDataset
+from project.models import AccessPolicy, DUASignature, DataAccessRequest, PublishedProject
+from user.models import Training, TrainingType
+
+
+def get_public_projects_query():
+    return Q(access_policy=AccessPolicy.OPEN)
+
+
+def get_restricted_projects_query(user):
+    dua_signatures = DUASignature.objects.filter(user=user)
+    query = Q(access_policy=AccessPolicy.RESTRICTED) & Q(duasignature__in=dua_signatures)
+    return query
+
+
+def get_credentialed_projects_query(user):
+    dua_signatures = DUASignature.objects.filter(user=user)
+    completed_training = (
+        Training.objects.get_valid()
+        .filter(user=user)
+        .values_list("training_type")
+    )
+    not_completed_training = TrainingType.objects.exclude(pk__in=completed_training)
+    required_training_complete = ~Q(required_trainings__in=not_completed_training)
+
+    accepted_data_access_requests = DataAccessRequest.objects.filter(
+        requester=user, status=DataAccessRequest.ACCEPT_REQUEST_VALUE
+    )
+
+    contributor_review_with_access = Q(
+        access_policy=AccessPolicy.CONTRIBUTOR_REVIEW
+    ) & Q(data_access_requests__in=accepted_data_access_requests)
+
+    credentialed_with_dua_signed = Q(
+        access_policy=AccessPolicy.CREDENTIALED
+    ) & Q(duasignature__in=dua_signatures)
+
+    query = required_training_complete & (
+        contributor_review_with_access | credentialed_with_dua_signed
+    )
+    return query
+
+
+def get_projects_accessible_through_events(user):
+    events_all = Event.objects.filter(Q(host=user) | Q(participants__user=user))
+    active_events = set(events_all.filter(end_date__gte=datetime.now()))
+    accessible_datasets = EventDataset.objects.filter(event__in=active_events, is_active=True)
+    accessible_projects_ids = []
+    for event_dataset in accessible_datasets:
+        if event_dataset.has_access(user):
+            accessible_projects_ids.append(event_dataset.dataset.id)
+    query = Q(id__in=accessible_projects_ids)
+    return query
+
+
+def get_accessible_projects(user):
+    """
+    Returns query filter for published projects accessible by a specified user
+    """
```

Comment:
Please could you try to use consistent formatting? This function has lots of white space. The previous function (`get_projects_accessible_through_events`) is dense!

Reply:
My bad. let me work on them. i will remember in future. Thank you